src/PythonPluginManager.cpp

```diff
@@ -20,14 +20,69 @@
 #include <QDirIterator>
 #include <QFileInfo>
 
+using namespace pybind11::literals;
+
 const std::vector<Runtime::RegisteredPlugin> &PythonPluginManager::plugins() const
 {
     return m_plugins;
 }
 
-// Loading python plugins works in 3 steps
+void PythonPluginManager::loadPluginsFromEntryPoints()
+{
+    plgPrint() << "Searching for custom plugins (checking metadata in site-package)";
+    py::object versionInfo = py::module::import("sys").attr("version_info");
+    py::object metadata;
+
+    if (versionInfo < py::make_tuple(3, 10))
+    {
+        metadata = py::module::import("importlib_metadata");
```

Comment:
This is a third party package installable from pip its not installed by default in environments 🤔

Reply:
Yes you are right I forgot about the fact that user can switch their venv! Maybe we can simply limit this feature to user of python > 3.10?

Reply:
Yes I think we can do that

Reply:
I think there is no way that user can use a venv with a different version than the one used to compile CCPyPlugin, it won't work right? This way we could maybe add a compile time check for the version of Python instead of the current runtime check added in this PR. but what do you prefer, runtime check or compile time check? One other solution is maybe to filter entry point ourselves with pybinds for py < 3.10, without relying on importlib_metadata, I do not know if it's feasible but I think there is no reason it should not.

Reply:
Yes using a different python version (major.minor) is not going to work, and this plugin code checks against that

If you can make it work by filtering yourself and that its not too complicated im fine with it.

Otherwise we will add a compile time check to only allow against python >= 3.10